src/System.Management.Automation/DscSupport/JsonDeserializer.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.PowerShell.DesiredStateConfiguration
+{
+    internal class JsonDeserializer
+    {
+        #region Constructors
+
+        /// <summary>
+        /// Instantiates a default deserializer
+        /// </summary>
+        public static JsonDeserializer Create()
+        {
+            return new JsonDeserializer();
+        }
+
+        #endregion Constructors
+
+        #region Methods
+
+        /// <summary>
+        /// Returns schema of Cim classes from specified json file
+        /// </summary>
+        public IEnumerable<PSObject> DeserializeClasses(string json)
```

Comment:
some parameter validation for public API should be done.

Reply:
Added check for IsNullOrEmpty.